### PR TITLE
Add power data to gps_parser and filter_gpstext

### PR DESCRIPTION
**Pull request by augustoborn**
Added power meter data to the GPS filter, tested and working on a GPX file extracted from Strava ([gpxexample.zip](https://github.com/user-attachments/files/16323523/gpxexample.zip)
).

### CODE REVIEW

**Comment by ddennedy** [comment]
@dany123 should review and approve this since he wrote that code.

File: src/modules/qt/gps_parser.cpp [hunk 1]
```diff
@@ -937,6 +940,22 @@ void process_gps_smoothing(gps_private_data gdata, char do_processing)
                 nr_atemp++;
             }
 
+            //power
+            if (gp_r[i].power != GPS_UNINIT) {
+                if (power != GPS_UNINIT && nr_power > 0 && nr_power <= nr_one_minute * 60) {
```
Comment (dany123) [inline src/modules/qt/gps_parser.cpp:945]: > nr_one_minute * 60

Not very important but I recommend removing the "* 60" part and just leave the one minute default allowed gap. While I see that Strava fills in with 0 during pauses, I can imagine some other service not bothering to write anything and you'd have a period of up to 1 hour of not pedaling where the power would be interpolated for no reason.

File: src/modules/qt/gps_parser.cpp [hunk 2]
```diff
@@ -1200,6 +1226,8 @@ void qxml_parse_tcx(QXmlStreamReader &reader, gps_point_ll **gps_list, int *coun
                     crt_point.cad = reader.readElementText().toDouble();
                 } else if (reader.name() == QString("Temperature")) {
                     crt_point.atemp = reader.readElementText().toDouble();
+                } else if (reader.name() == QString("Power")) { // guessing
```
Comment (dany123) [inline src/modules/qt/gps_parser.cpp:1229]: A Shotcut forum user added a [sample tcx ](https://forum.shotcut.org/t/gps-text-video-filter/29278/34?u=daniel47) when requesting power to be added, it seems the Garmin schema uses this (a bit more complex) extension for storing power:

            <Extensions>
              <ns3:TPX>
                <ns3:Speed>0.0</ns3:Speed>
                <ns3:Watts>0</ns3:Watts>
              </ns3:TPX>
            </Extensions>

File: src/modules/qt/filter_gpstext.yml
Comment (dany123) [inline src/modules/qt/filter_gpstext.yml:1]: I think the version (on line 5) should be incremented.

File: src/modules/qt/gps_parser.cpp (hunk 2 above)
Comment (dany123) [reply src/modules/qt/gps_parser.cpp:1229]: FYI, if the namespace thing is too annoying to deal with or you don't care about the .gpx section I am fine with approoving this as is.

File: src/modules/qt/filter_gpstext.yml
Comment (ddennedy) [reply src/modules/qt/filter_gpstext.yml:1]: Will there be a UI change in Shotcut or Kdenlive that needs to depend on the version?

File: src/modules/qt/gps_parser.cpp (hunk 2 above)
Comment (ddennedy) [reply src/modules/qt/gps_parser.cpp:1229]: Garmin is very popular, but we can merge this and then try to extend to tcx. I am getting the impression that we are not going to hear back from the contributor.

File: src/modules/qt/filter_gpstext.yml
Comment (frdbr) [reply src/modules/qt/filter_gpstext.yml:1]: FYI: Kdenlive does not have this feature implemented.

Comment (dany123) [reply src/modules/qt/filter_gpstext.yml:1]: > Will there be a UI change in Shotcut 

For completeness  there should be a new value in the drop-down list of available fields. 
I can make a quick PR for it tomorrow.

**Pull request merged**